spec/requests/api/v2/providers/sites_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'rails_helper'
+
+describe 'Sites API v2', type: :request do
+  let(:user) { create(:user) }
+  let(:organisation) { create(:organisation, users: [user]) }
+  let(:payload) { { email: user.email } }
+  let(:token) { build_jwt :apiv2, payload: payload }
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  let(:site1) { create :site, location_name: 'Main site 1' }
+  let(:site2) { create :site, location_name: 'Main site 2' }
+  let(:sites) { [site1, site2] }
+
+  let!(:provider) {
+    create(:provider,
+           course_count: 0,
+           site_count: 0,
+           sites: sites,
+           organisations: [organisation])
+  }
+
+  subject { response }
+
+  describe 'GET show' do
+    let(:show_path) do
+      "/api/v2/providers/#{provider.provider_code}" +
+        "/sites/#{site1.id}"
```

Comment:
Just had a thought about `site.id` . I wonder whether we may need to use campus code here, like we use provider code and campus code for the frontend URLs. I think @fofr’s prototype uses these in the URL for the site show/edit page. 

This introduces complications because the campus code data is inconsistent ie some main site campus codes are `-` and others are an empty string ` `, which would need translation in the URL :-(

Not 100% sure what the right answer is

Reply:
Exactly that @benilovj. We don't have unique `location_name`'s at present so I went with `id` to be safe.

Reply:
Ok lets go with that for now, can easily tweak if needed.